libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -514,70 +517,105 @@ protected async Task<DialogTurnResult> ContinueActionsAsync(DialogContext dc, ob
             // from the stack and we want to detect this so we can stop processing actions.
             var instanceId = GetUniqueInstanceId(actionContext);
 
-            // Execute queued actions
-            var actionDC = CreateChildContext(actionContext);
-            while (actionDC != null)
+            try
             {
-                // Continue current step
-                // DEBUG: To debug step execution set a breakpoint on line below and add a watch 
-                //        statement for actionContext.Actions.
-                var result = await actionDC.ContinueDialogAsync(cancellationToken).ConfigureAwait(false);
+                OnPushScopedServices(dc.Context);
 
-                // Start step if not continued
-                if (result.Status == DialogTurnStatus.Empty && GetUniqueInstanceId(actionContext) == instanceId)
+                // Execute queued actions
+                var actionDC = CreateChildContext(actionContext);
+                while (actionDC != null)
                 {
-                    // Call begin dialog on our next step, passing the effective options we computed
-                    var nextAction = actionContext.Actions.First();
-                    result = await actionDC.BeginDialogAsync(nextAction.DialogId, nextAction.Options, cancellationToken).ConfigureAwait(false);
-                }
+                    // Continue current step
+                    // DEBUG: To debug step execution set a breakpoint on line below and add a watch 
+                    //        statement for actionContext.Actions.
+                    var result = await actionDC.ContinueDialogAsync(cancellationToken).ConfigureAwait(false);
 
-                // Is the step waiting for input or were we cancelled?
-                if (result.Status == DialogTurnStatus.Waiting || GetUniqueInstanceId(actionContext) != instanceId)
-                {
-                    return result;
-                }
+                    // Start step if not continued
+                    if (result.Status == DialogTurnStatus.Empty && GetUniqueInstanceId(actionContext) == instanceId)
+                    {
+                        // Call begin dialog on our next step, passing the effective options we computed
+                        var nextAction = actionContext.Actions.First();
```

Comment:
can you guarantee that there is an action in the Actions collection here?

Reply:
yeah, actioncontext creates empty actions

---
In reply to: [394775062](https://github.com/microsoft/botbuilder-dotnet/pull/3594#discussion_r394775062) [](ancestors = 394775062)